src/widgets/stadiumlist.cpp

```diff
@@ -0,0 +1,128 @@
+#include "stadiumlist.hpp"
+
+/**
+ * This constructor constructs a new StadiumList inside of the parent widget and populates it with
+ * the information from the database
+ * @param parent pointer to parent widget
+ */
+StadiumList::StadiumList(QWidget *parent) :
+    QWidget(parent)
+{
+    //creates new tree widget
+    m_listDisplay = new QTreeWidget(this);
+    m_listDisplay->resize(parent->size());
+    //populates the tree widget from the database
+    populateWidget(Database::getTeamsAndStadiums());
+    //sets up the stadium clicked signal
+    QObject::connect(m_listDisplay, SIGNAL(itemClicked(QTreeWidgetItem*, int)),this, SLOT(connectWidgetItemToStadium(QTreeWidgetItem*)));
+}
+
+/**
+ * This constructor constructs a new Stadium List inside of the parent widget and populates it
+ * with the information from the given vector
+ * @param stadiumList vector containing information to populate Stadium List with
+ * @param parent pointer to the parent widet
+ */
+StadiumList::StadiumList(const std::vector<std::pair<Team,Stadium>>& stadiumList, QWidget *parent) :
+    QWidget(parent)
+{
+    //creates new tree widget
+    m_listDisplay = new QTreeWidget(this);
+    m_listDisplay->resize(parent->size());
+    //populates the widget from the vector of teams and stadiums
+    populateWidget(stadiumList);
+    //sets up the stadium clicked signal
+    QObject::connect(m_listDisplay, SIGNAL(itemClicked(QTreeWidgetItem*, int)),this, SLOT(connectWidgetItemToStadium(QTreeWidgetItem*)));
+}
+
+/**
+ * setStadiumTeamList resets the tree widget and populates it with the information from the vector passed in
+ * @param stadiumList information to repopulate the tree widget with
+ */
+void StadiumList::setStadiumTeamList(const std::vector<std::pair<Team,Stadium>>& stadiumList)
```

Comment:
Not sure you why have this function if it just calls `populateWidget()`, redundant

Reply:
When the vector was there, it has meaning, but I forgot to take it out